File: python/affiche.py
import os
import signal

print(f"Mon PID est : {os.getpid()}")
print("En attente d'un signal...")
signal.pause()  # il va attendre un signal

f = open("/tmp/entier.txt", "r")
nombre = int(f.read())
f.close()
print(f"Entier recu : {nombre}")

File: C/affiche.c
#include <stdio.h>
#include <unistd.h>
#include <signal.h>

int main()
{
    int nombre;
    FILE *f;
    
    printf("Mon PID est : %d\n", getpid());
    printf("En attente d'un signal...\n");   
    pause();  // il va attendr un signal
    
    f=fopen("/tmp/entier.txt", "r");
    fscanf(f, "%d", &nombre);
    fclose(f);   
    printf("Entier recu : %d\n", nombre);
    return 0;
}